src/panels/lovelace/header-footer/structs.ts

```diff
@@ -22,8 +23,22 @@ export const graphHeaderFooterConfigStruct = object({
   hours_to_show: optional(number()),
 });
 
-export const headerFooterConfigStructs = union([
-  pictureHeaderFooterConfigStruct,
-  buttonsHeaderFooterConfigStruct,
-  graphHeaderFooterConfigStruct,
-]);
+export const headerFooterConfigStructs = dynamic<any>((value) => {
+  if (value && typeof value === "object" && "type" in value) {
+    switch ((value as LovelaceHeaderFooterConfig).type!) {
+      case "buttons": {
+        return buttonsHeaderFooterConfigStruct;
+      }
+      case "graph": {
+        return graphHeaderFooterConfigStruct;
+      }
+      case "picture": {
+        return pictureHeaderFooterConfigStruct;
+      }
+    }
+  }
+
+  // No "type" property => we fallback to one random variant, which ensure that user gets informed
+  // about missing "type", as all variants have that marked as required.
+  return pictureHeaderFooterConfigStruct;
```

Comment:
Why not fallback to all of them?

Reply:
union([buttonsHeaderFooterConfigStruct, graphHeaderFooterConfigStruct, pictureHeaderFooterConfigStruct]);